nixos/modules/services/backup/bupstash.nix

```diff
@@ -0,0 +1,378 @@
+        type = bool;
+        default = true;
+        description = "If set to true, restricts the commands that can be run through ssh by the user to `bupstash ...`.";
+      };
+    };
+
+    jobs = mkOption {
+      type = attrsOf (
+        submodule (
+          { name, config, ... }:
+          {
+            options = {
+              paths = mkOption {
+                type = listOf path;
+                default = [ ];
+                description = "List of paths to backup.";
+              };
+
+              excludes = mkOption {
+                type = listOf str;
+                default = [ ];
+                description = "List of patterns to exlude from the backup.";
+              };
+
+              command = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "Command used for getting the backup data.";
+              };
+
+              key = mkOption {
+                type = path;
+                description = "A path to the key used to backup the data.";
+              };
+
+              user = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "User for the backup job.";
+              };
+
+              group = mkOption {
+                type = nullOr str;
+                default = null;
+                description = "Group for the backup job.";
+              };
+
+              preHook = mkOption {
+                type = lines;
+                default = "";
+                description = "Commands to run before `bupstash put`.";
+              };
+
+              postHook = mkOption {
+                type = lines;
+                default = "";
+                description = "Commands to run after `bupstash put`.";
+              };
+
+              repositories = mkOption {
+                type = listOf (
+                  either str (
+                    submodule {
+                      options = {
+                        user = mkOption {
+                          type = str;
+                          description = "User used to connect to the remote server.";
+                        };
+
+                        server = mkOption {
+                          type = str;
+                          description = "Address ot the remote server.";
+                        };
+
+                        path = mkOption {
+                          type = nullOr str;
+                          description = "Remote path for the repository.";
+                        };
+
+                        privateKey = mkOption {
+                          type = path;
+                          description = "Private SSH key used to connect to the server.";
+                        };
+                      };
+                    }
+                  )
+                );
+                default = [ ];
+                description = "List of repositories to backup to.";
```

Comment:
I noticed, by chance, that I can pass a path to point to a local directory, mention that in the description.

Reply:
I talked with @piegamesde yesterday, and we concluded that the whole repository options is too confining. His upcoming review will likely touch on this subject as well.

Reply:
Described in the repository `kind` option